language-plutus-core/generators/Language/PlutusCore/Generators/Internal/Dependent.hs

```diff
@@ -25,31 +25,27 @@ liftOrdering GT = GGT
 instance GEq TypedBuiltinSized where
     TypedBuiltinSizedInt  `geq` TypedBuiltinSizedInt  = Just Refl
     TypedBuiltinSizedBS   `geq` TypedBuiltinSizedBS   = Just Refl
-    TypedBuiltinSizedSize `geq` TypedBuiltinSizedSize = Just Refl
     _                     `geq` _                     = Nothing
 
-instance (Pretty size, Eq size) => GEq (TypedBuiltin size) where
-    TypedBuiltinSized size1 tbs1 `geq` TypedBuiltinSized size2 tbs2 = do
-        guard $ size1 == size2
-        tbs1 `geq` tbs2
+instance GEq TypedBuiltin where
+    TypedBuiltinSized tbs1 `geq` TypedBuiltinSized tbs2 = tbs1 `geq` tbs2
```

Comment:
> I guess we no longer need `TypedBuiltinSized` and can just inline its constructors into `TypedBuiltin`.

No, don't do that. Now that we do not have sizes, I can make them interoperate properly with known types, so I'll do something myself.

Reply:
This sort of stuff is what I was planning to leave for later - I think it's probably better to just do a pass over this later and simplify, rather than me trying to do too much refactoring at the same time.